### PR TITLE
Customisation

### PR DESCRIPTION
**Pull request by maleck13**
## Motivation
https://trello.com/c/Olk3Fnq7/234-8-installer-customisation

## What
a new inventory added to a secret in the cluster to allow customisation

## Why
to allow external parties to add customisations they need for things like workshops

## Verification Steps
clone repo
run installer
check the secret is created with the values expected
``` oc get secret manifest -n webapp --template '{{index .data "generated_inventory"}}'  | base64 -D ```

## Progress
- [x] Complete custom inventory file
- [x] Finish doc Todo

### CODE REVIEW

File: docs/customisation.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Customising an Integreatly Cluster
+
```
Comment (maleck13) [inline docs/customisation.md]: Ignore this doc it is still in progress

**Comment by maleck13** [comment]
@pmccarthy  ok to take a look @aidenkeating

**Comment by maleck13** [comment]
@pmccarthy If we feel we get enough value from this as is without the rest of the card, I say we go ahead and merge once verified and progress the manifest versions and optional service installation separately

File: evals/roles/customisation/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+- name: Retrieve launcher sso hostvars
+  shell: "oc get route/{{launcher_sso_prefix}} -o jsonpath='{.spec.host}' -n {{ launcher_namespace }}"
+  register: launcher_sso_host_cmd
+
+- set_fact: 
+    launcher_sso_route: "{{launcher_sso_host_cmd.stdout}}"          
+
+
+
+- name: Generate custom inventory
```
Comment (pmccarthy) [inline evals/roles/customisation/tasks/main.yaml]: Might be worth adding a task to clean up the generated inventory file once the secret has been created

File: evals/roles/customisation/templates/inventory.j2 [hunk 3]
```diff
@@ -0,0 +1,109 @@
+[local:vars]
+ansible_connection=local
+
+# The openshift apps suffix
+ocp_apps_domain="{{eval_app_host}}"
```
Comment (pmccarthy) [inline evals/roles/customisation/templates/inventory.j2]: I think we should rename `ocp_apps_domain` to something like `ocp_routing_subdomain` as this is how it's referenced in the Openshift docs [1] Apps domain may cause confusion.

[1] https://docs.openshift.com/container-platform/3.11/install_config/router/default_haproxy_router.html#customizing-the-default-routing-subdomain

File: evals/roles/customisation/templates/inventory.j2 [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+seed_user_count="{{eval_seed_users_count}}"
```
Comment (pmccarthy) [inline evals/roles/customisation/templates/inventory.j2]: Is `seed_user_count` required since we already have `rhsso_seed_users_count` listed?

**Comment by pmccarthy** [comment]
Verifying changes against PDS (v3.10 cluster)

**Comment by pmccarthy** [comment]
Ran installer against PDS and attempted to get the secret. The command documented to get the secret failed as the `-D` param for base64 is not recognised. Changed to lower case `-d` and it worked.

The PDS clusters run RHEL v7.5

**Comment by maleck13** [comment]
@pmccarthy yeah mac has ```-D``` and linux ```-d``` will update the doc

File: evals/roles/customisation/templates/inventory.j2 (hunk 4 above)
Comment (maleck13) [reply evals/roles/customisation/templates/inventory.j2]: wasn't sure on that one. But we seem to have different vars for each?

File: evals/roles/customisation/templates/inventory.j2 (hunk 3 above)
Comment (maleck13) [reply evals/roles/customisation/templates/inventory.j2]: sure sounds good

**Comment by maleck13** [comment]
@pmccarthy updated based on review. Let me know if you are happy to approve

File: evals/roles/customisation/templates/inventory.j2 (hunk 4 above)
Comment (pmccarthy) [reply evals/roles/customisation/templates/inventory.j2]: Had a second look at the code and it looks like the `rhsso_seed_users_count` gets set to whatever the value of `eval_seed_users_count` is. I think to be safe we should probably leave both for now.

**Comment by camilamacedo86** [comment]
@maleck13 this PR has sintax errors. 

```shell
$ ansible-playbook -i inventories/hosts playbooks/install.yml -e github_client_id=fd300e7e29fc2ed7bb49 -e github_client_secret=19efea317c67b416b536e26f7627b89eecf3fc2e 
ERROR! Syntax Error while loading YAML.
  mapping values are not allowed in this context

The error appears to have been in '/Users/camilamacedo/integr8ly/installation/evals/roles/customisation/tasks/main.yaml': line 98, column 9, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

- name: remove generated template
    file: path='/tmp/generated_inventory' state=absent
        ^ here

```

**Comment by maleck13** [comment]
@camilamacedo86 thanks for spotting this, I will add a fix

**Pull request merged**